src/modules/navigator/GeofenceBreachAvoidance/fake_geofence.hpp

```diff
@@ -0,0 +1,183 @@
+
+
+
+#include"../geofence.h"
+#include <lib/mathlib/mathlib.h>
+
+class FakeGeofence : public Geofence
+{
+public:
+	FakeGeofence() :
+		Geofence(nullptr)
+	{};
+
+	virtual ~FakeGeofence() {};
+
+	virtual bool isInsidePolygonOrCircle(double lat, double lon, float altitude)
+	{
+		switch (_probe_function_behavior) {
+		case ALL_POINTS_OUTSIDE: {
+				return _allPointsOutside(lat, lon, altitude);
+			}
+
+		case LEFT_INSIDE_RIGHT_OUTSIDE: {
+				return _left_inside_right_outside(lat, lon, altitude);
+			}
+
+		case RIGHT_INSIDE_LEFT_OUTSIDE: {
+				return _right_inside_left_outside(lat, lon, altitude);
+			}
+
+		case GF_BOUNDARY_20M_AHEAD: {
+				return _gf_boundary_is_20m_ahead(lat, lon, altitude);
+			}
+
+		default:
+			return _allPointsOutside(lat, lon, altitude);
+		}
+	}
+
+	enum PROBE_FUNC_ENUM {
+		ALL_POINTS_OUTSIDE = 0,
+		LEFT_INSIDE_RIGHT_OUTSIDE,
+		RIGHT_INSIDE_LEFT_OUTSIDE,
+		GF_BOUNDARY_20M_AHEAD
+	};
+
+	void setProbeFunctionBehavior(PROBE_FUNC_ENUM func) {_probe_function_behavior = func;}
+
+
+private:
+
+	PROBE_FUNC_ENUM _probe_function_behavior = ALL_POINTS_OUTSIDE;
+
+	bool _allPointsOutside(double lat, double lon, float alt)
+	{
+		return false;
+	}
+
+	bool _left_inside_right_outside(double lat, double lon, float alt)
+	{
+		static int flag = true;
+
+		if (flag) {
+			flag = false;
+			return true;
+
+		} else {
+			return false;
+		}
+	}
+
+	bool _right_inside_left_outside(double lat, double lon, float alt)
+	{
+		static int flag = false;
+
+		if (flag) {
+			flag = false;
+			return true;
+
+		} else {
+			flag = true;
+			return false;
+		}
+	}
+
+	bool _gf_boundary_is_20m_ahead(double lat, double lon, float alt)
```

Comment:
Might be a detail, but this only does what I'd expect from the method name if yaw=0.

Reply:
renamed to `XXX_north`